src/libraries/System.Runtime/tests/System/DoubleTests.cs

```diff
@@ -351,6 +352,58 @@ public static void Parse(string value, NumberStyles style, IFormatProvider provi
             }
         }
 
+        internal static string SplitPairs(string input)
```

Comment:
I could optimize this method and `Reverse` to be a little less allocate-y if we really wanted to, but I don't really care much about them since they are just unit tests